app/src/ui/toolbar/push-pull-button.tsx

```diff
@@ -265,9 +265,41 @@ export class PushPullButton extends React.Component<
     this.props.dispatcher.push(this.props.repository)
   }
 
+  /**
+   * This is a hack to get the screen reader to read the message after the force
+   * push confirm dialog closes.
+   *
+   * Problem: The dialog component sets the focus back to what ever was in the
+   * `document.ActiveElement` when the dialog was opened. However the active
+   * element is the force push button that is replaced with the fetch button.
+   * Thus, the force push element is no longer present when the dialog closes
+   * and the focus defaults to the document body. This means the sr message is
+   * not read.
+   *
+   * Solution: Set the `document.ActiveElement` to an element containing the sr
+   * element before opening the dialog so that it returns the focus to an
+   * element containing the sr. You can do this by calling the `focus` element
+   * of a tab focusable element hence adding the tab index.
+   *
+   * Other notes: If I set the focus to the sr element directly, it causes the
+   * message to be read twice.
+   */
+  private setScreenReaderStateMessageFocus() {
+    const srElement = document.getElementById('push-pull-button-state')
+    if (srElement) {
+      srElement.tabIndex = -1
+      srElement.focus()
+    }
+  }
+
   private forcePushWithLease = () => {
     this.closeDropdown()
-    this.props.dispatcher.confirmOrForcePush(this.props.repository)
+
+    setTimeout(() => {
```

Comment:
So this is the only thing I could get to work... I think I found the cause as to why the dropdown causes a timing issue and that is because the dropdown is in a `FocusTrap` and it has some logic like a `dialog` does and returns focus.. However, it doesn't use the `document.ActiveElement` to do so. The `FocusTrapOptions` allowed me to tell it to not return focus to the activating element (the dropdown button).. and it appeared it worked in such that when I was console logging out the active element change, it no longer returned focus to that button.. but there must be _something_ where it interrupts my hack from working... thinking possible a `blur()` event.. causing the `document.ActiveElement` to become set to default... but I don't know how to confirm that in the `FocusTrap` library stuff.

Thus currently this `forcePushWithLease` is calling `this.closeDropdown()` which triggers the dropdown menu to close when triggers the `FocusTrap` deactivation logic. Then, the `setTimeout` just causes the setting of the focus to happen on the next cycle of events.. which is evidently is after the `FocusTrap` logic has finished up.

Reply:
If you have any better ideas I am all for it.. as this just got hackier :D

Reply:
I also tried setting the focus to the message thing with the `FocusTrap`'s `onDeactivate` method.. but seems it still returned focus to the dropdown button.

Reply:
And.. on typing that I realized I didn't try the `onDeactivate` in combination `returnFocusOnDeactivate: false,`... and that works... but now got to figure out the proper way to tell it when to do that.

Reply:
I got that to work - so don't need timeout.. tho the hack got a little more involved.